akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -61,6 +61,24 @@ abstract class GraphStage[S <: Shape] extends GraphStageWithMaterializedValue[S,
   def createLogic: GraphStageLogic
 }
 
+/**
+ * A SourceStage represents a reusable graph stream processing stage. A SourceStage consists of a [[akka.stream.Shape]] which describes
+ * its output port.
+ */
+abstract class SourceStage[Out, M](name: String) extends GraphStageWithMaterializedValue[SourceShape[Out], M] {
```

Comment:
I would add convenience methods, `push(elem) ` == `push(out, elem)` and similarly for `pull()`